### PR TITLE
Input data fixes

### PR DESCRIPTION
**Pull request by Clare2D**
Here are some changed that I was working on prior to the current_web_functionality being merged. Unfortunately I overwrote the spacing changes that had been made to the files. Sorry about this.

### CODE REVIEW

File: web_tool_script_2.R [hunk 1]
```diff
@@ -168,47 +160,31 @@ if (file.exists(bonds_inputs_file)) {
...
-    }
+    if(data_check(company_all_cb)){ write_rds(company_all_cb, paste0(pf_file_results_path, "Bonds_results_company.rda"))}
+    if(data_check(port_all_cb)){write_rds(port_all_cb, paste0(pf_file_results_path, "Bonds_results_portfolio.rda"))}
+    if(has_map){if(data_check(map_cb)){write_rds(map_cb, paste0(pf_file_results_path, "Bonds_results_map.rda"))}}
   }
 }
 
-invisible(set_portfolio_parameters(file_path = fs::path(par_file_path, paste0(portfolio_name_ref_all, "_PortfolioParameters.yml"))))
-
-source(paste0(stress_test_path, "web_tool_stress_test.R"))
```
[the remark this answers is not in the document]
Comment (Clare2D) [reply web_tool_script_2.R:214]: Sorry I missed those ones!!!

[the remark this answers is not in the document]
Comment (Clare2D) [reply web_tool_script_2.R:214]: (also I pressed resolve without meaning too... I didn't resolve things)

[the remark this answers is not in the document]
Comment (Clare2D) [reply web_tool_script_2.R:214]: Done

**Review by jacobvjk — COMMENTED** [review]
Stylewise I am a bit reluctant to merging this PR because there are so many small changes that make the code overall less visually consistent. That makes it hard to fully understand the intended changes.

Other than that: lgtm

File: 0_portfolio_input_check_functions.R [hunk 2]
```diff
@@ -244,7 +243,7 @@ map_security_sectors <- function(fin_data, sector_bridge) {
   fin_data %>%
     group_by(security_mapped_sector) %>%
     filter(is.na(security_mapped_sector)) %>%
-    summarise(count = n())
+    summarise(count = n(), .groups = "drop")
```
Comment (maurolepore) [inline 0_portfolio_input_check_functions.R:246]: I see

```r
Quitting from lines 251-264 (integration-test.Rmd) 
Error: [conflicted] `summarize` found in 2 packages.
Either pick the one you want with `::` 
* plyr::summarize
* dplyr::summarize
Or declare a preference with `conflict_prefer()`
* conflict_prefer("summarize", "plyr")
* conflict_prefer("summarize", "dplyr")
```

but this error should go away after we merge https://github.com/2DegreesInvesting/PACTA_analysis/pull/179

File: 0_portfolio_input_check_functions.R [hunk 3]
```diff
@@ -254,7 +253,7 @@ map_security_sectors <- function(fin_data, sector_bridge) {
   return(fin_data)
 }
 
-map_comp_sectors <- function(comp_fin_data, sector_bridge) {
+map_comp_sectors <- function(comp_fin_data, sector_bridge, sector_bridge_override) {
```
Comment (maurolepore) [inline 0_portfolio_input_check_functions.R:256]: Changes to the signature of a function are risky. I recommend writing some kind of test.

File: 0_portfolio_input_check_functions.R [hunk 4]
```diff
@@ -280,10 +279,18 @@ map_comp_sectors <- function(comp_fin_data, sector_bridge) {
   comp_fin_data %>%
     group_by(mapped_sector) %>%
     filter(is.na(mapped_sector)) %>%
-    summarise(count = n())
+    summarise(count = n(), .groups = "drop")
```
Comment (maurolepore) [inline 0_portfolio_input_check_functions.R:282]: :+1:

File: 0_portfolio_input_check_functions.R [hunk 5]
```diff
@@ -349,9 +356,9 @@ check_asset_types <- function(fin_data) {
   fin_data$asset_type <- first_char_up(fin_data$asset_type)
 
   ### TEST
-  if (!any(unique(fin_data$asset_type) %in% allowable_asset_list())) {
-    stop("Check Financial Data Asset Types")
-  }
+  # if (!any(unique(fin_data$asset_type) %in% allowable_asset_list())){
+  #   stop("Check Financial Data Asset Types")
+  # }
```
Comment (maurolepore) [inline 0_portfolio_input_check_functions.R:361]: I suggest you go ahead and delete dead code.

File: 0_portfolio_input_check_functions.R [hunk 6]
```diff
@@ -1140,7 +1150,7 @@ create_merged_portfolio <- function(eq_portfolio, cb_portfolio) {
   return(portfolio_)
 }
 
-create_portfolio_subset <- function(portfolio, portfolio_type, relevant_fin_data) {
+create_portfolio_subset <- function(portfolio, portfolio_type) {
```
Comment (maurolepore) [inline 0_portfolio_input_check_functions.R:1153]: Changes to the interface of a function are risky. Try to provide a test.

File: 0_portfolio_input_check_functions.R [hunk 7]
```diff
@@ -1324,7 +1326,7 @@ prepare_portfolio_emissions <- function(
       company_id,
       bics_sector,
       bics_subgroup,
-      mapped_sector
+      # mapped_sector
```
Comment (maurolepore) [inline 0_portfolio_input_check_functions.R:1329]: Can you delete this?

**Review by maurolepore — COMMENTED** [review]
* I removed the call to `save_file()` because that function no longer exists. 

* I styled the code with 

```r
r <- fs::dir_ls(regexp = "[.]R$", recurse = TRUE)
rmd <- fs::dir_ls(regexp = "[.]Rmd$", recurse = TRUE)
purrr::walk(c(r, rmd), styler::style_file)
```

You can use an RStudio addin to style each file.



-- 

Nit: Delete (rather than comment-out) dead code.

**Review by maurolepore — CHANGES_REQUESTED** [review]
I'd like to test this PR locally again after we merge #179

**Comment by maurolepore** [comment]
> Unfortunately I overwrote the spacing changes that had been made to the files. Sorry about this.

No problem. We do that automatically with the stlyer package.

File: 0_portfolio_input_check_functions.R (hunk 2 above)
Comment (maurolepore) [reply 0_portfolio_input_check_functions.R:246]: Actually, this is a different conflict. Do we need plyr at all? It has been superseded by dplyr for a long time.

**Comment by Clare2D** [comment]
Yeah I got these plyr errors as well. It is not needed from my perspective.
Thanks for the comments.

**Review by maurolepore — COMMENTED** [review]
The integration tests now passes locally, and the few things that remain unresolved are Nit.
So from the technical side  it LGTM.

The first comment of this conversation says:
> Here are some changed that I was working on prior to the current_web_functionality being merged.

@Clare2D or @jacobvjk, can you explain how this PR improves the code base?

**Comment by maurolepore** [comment]
Great! Thanks. I suggest you submit each intention in a separate PR. See @cjyetman  PRs -- they are a great example of bit-size prs. That helps keep each PR focused and allows the team to move faster --- as the reviewers are more likely to review a small, easy-to-understand PR than a big one.

**Comment by cjyetman** [comment]
@Clare2D I think this was superseded by #225 and #226. Is that correct?

**Comment by Clare2D** [comment]
Yeah this can be dropped

**Pull request closed without merging**